docker/image.go

```diff
@@ -0,0 +1,108 @@
+package docker
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/deis/deisrel/git"
+)
+
+// ErrInvalidImageName is the error returned when a func couldn't parse a string into an Image struct
+type ErrInvalidImageName struct {
+	Str string
+}
+
+// Error is the error interface implementation
+func (e ErrInvalidImageName) Error() string {
+	return fmt.Sprintf("%s is an invalid image name", e.Str)
+}
+
+// Image represents a single image name, including all information about its registry, repository and tag
+type Image struct {
+	registry string
+	repo     string
+	name     string
+	tag      string
+}
+
+// ParseImageFromName parses a raw image name string into an Image
+func ParseImageFromName(name string) (*Image, error) {
+	spl := strings.Split(name, "/")
+	splLast := strings.Split(spl[len(spl)-1], ":")
+	tag := "latest"
+	if len(splLast) > 1 {
+		tag = splLast[1]
+		spl[len(spl)-1] = splLast[0]
+	}
+	if len(spl) == 1 {
+		// dockerhub trusted image
+		return &Image{
+			registry: "",
+			repo:     "",
+			name:     spl[0],
+			tag:      tag,
+		}, nil
+	} else if len(spl) == 2 {
+		// dockerhub image
+		return &Image{
+			registry: "",
+			repo:     spl[0],
+			name:     spl[1],
+			tag:      tag,
+		}, nil
+	} else if len(spl) == 3 {
+		// non-dockerhub image
+		return &Image{
+			registry: spl[0],
+			repo:     spl[1],
+			name:     spl[2],
+			tag:      tag,
+		}, nil
+	}
+	return nil, ErrInvalidImageName{Str: name}
+}
+
+// ParseImageFromRepoAndSha attempts to convert ras into a docker image, using dockerRegistryOrg as the docker registry
+func ParseImageFromRepoAndSha(dockerRegistryOrg string, ras git.RepoAndSha) (*Image, error) {
+	str := fmt.Sprintf("quay.io/%s/%s:git-%s", dockerRegistryOrg, ras.Name, ras.ShortSHA())
```

Comment:
Does hard-coding this prevent us from retagging/pushing release images to the DockerHub registry?  It doesn't _appear_ that we're set up to push release images to both `quay.io` and `DockerHub` registries, but please correct me if I'm wrong.  Otherwise, definitely okay with covering in follow-up ticket.